dartagnan/src/main/java/com/dat3m/dartagnan/program/memory/Memory.java

```diff
@@ -16,9 +16,9 @@ public class Memory {
      * @return
      * Points to the created location.
      */
-    public MemoryObject allocate(int size, boolean isStatic) {
+    public MemoryObject allocate(int size, boolean isStatic, boolean virtuality) {
```

Comment:
`virtual` seems like a better name. Same applies to class `MemoryObject`

Reply:
Accepted

Reply:
Methods `setVirtuality` and `getVirtuality` also need to be updated

Reply:
Updated

Reply:
Actually, it seems the `virtual` field is only set when the object is initialized. Thus you should mark the field as `final` and remove the setter.

Reply:
Updated